code/modules/projectiles/projectile/experimental.dm

```diff
@@ -0,0 +1,644 @@
+	var/turf/T2 = get_turf(original)
+	var/X = T2.x - T1.x
+	var/Y = T2.y - T1.y
+	var/X_spawn = 0
+	var/Y_spawn = 0
+	if(X>0)
+		if(Y>0)
+			if(X>Y)
+				pos_from = WEST
+				pos_to = NORTH
+				X_spawn = 1
+			else if(X<Y)
+				pos_from = SOUTH
+				pos_to = EAST
+				Y_spawn = 1
+			else
+				if(prob(50))
+					pos_from = WEST
+					pos_to = NORTH
+					X_spawn = 1
+				else
+					pos_from = SOUTH
+					pos_to = EAST
+					Y_spawn = 1
+		else if(Y<0)
+			if(X>(Y*-1))
+				pos_from = WEST
+				pos_to = SOUTH
+				X_spawn = 1
+			else if(X<(Y*-1))
+				pos_from = NORTH
+				pos_to = EAST
+				Y_spawn = -1
+			else
+				if(prob(50))
+					pos_from = WEST
+					pos_to = SOUTH
+					X_spawn = 1
+				else
+					pos_from = NORTH
+					pos_to = EAST
+					Y_spawn = -1
+		else if(Y==0)
+			pos_from = WEST
+			X_spawn = 1
+			if(prob(50))
+				pos_to = NORTH
+			else
+				pos_to = SOUTH
+	else if(X<0)
+		if(Y>0)
+			if((X*-1)>Y)
+				pos_from = EAST
+				pos_to = NORTH
+				X_spawn = -1
+			else if((X*-1)<Y)
+				pos_from = SOUTH
+				pos_to = WEST
+				Y_spawn = 1
+			else
+				if(prob(50))
+					pos_from = EAST
+					pos_to = NORTH
+					X_spawn = -1
+				else
+					pos_from = SOUTH
+					pos_to = WEST
+					Y_spawn = 1
+		else if(Y<0)
+			if((X*-1)>(Y*-1))
+				pos_from = EAST
+				pos_to = SOUTH
+				X_spawn = -1
+			else if((X*-1)<(Y*-1))
+				pos_from = NORTH
+				pos_to = WEST
+				Y_spawn = -1
+			else
+				if(prob(50))
+					pos_from = EAST
+					pos_to = SOUTH
+					X_spawn = -1
+				else
+					pos_from = NORTH
+					pos_to = WEST
+					Y_spawn = -1
+		else if(Y==0)
+			pos_from = EAST
+			X_spawn = -1
+			if(prob(50))
+				pos_to = NORTH
+			else
+				pos_to = SOUTH
+	else if(X==0)
+		if(Y>0)
+			Y_spawn = 1
+			pos_from = SOUTH
+			if(prob(50))
+				pos_to = EAST
+			else
+				pos_to = WEST
+		else if(Y<0)
+			Y_spawn = -1
+			pos_from = NORTH
+			if(prob(50))
+				pos_to = EAST
+			else
+				pos_to = WEST
+	else
+		OnDeath()
+		loc = null
+		returnToPool(src)
+		return
+
+	var/turf/newspawn = locate(T1.x + X_spawn, T1.y + Y_spawn, z)
+	src.loc = newspawn
+
+	update_icon()
+
+/obj/item/projectile/ricochet/update_icon()//8 possible combinations
+	switch(pos_to)
+		if(NORTH)
+			if(pos_from == WEST)
+				dir = NORTHWEST
+			else
+				dir = EAST
+		if(SOUTH)
+			if(pos_from == WEST)
+				dir = WEST
+			else
+				dir = SOUTHEAST
+		if(EAST)
+			if(pos_from == NORTH)
+				dir = NORTHEAST
+			else
+				dir = SOUTH
+		if(WEST)
+			if(pos_from == NORTH)
+				dir = NORTH
+			else
+				dir = SOUTHWEST
+
+/obj/item/projectile/ricochet/Bump(atom/A as mob|obj|turf|area)
+	if(bumped)	return 0
+	bumped = 1
+
+	if(A)
+		if(istype(A,/turf/) || (istype(A,/obj/machinery/door/) && A.opacity))
+			var/obj/structure/ricochet_bump/bump = new(loc)
+			bump.dir = pos_to
+			playsound(get_turf(src), 'sound/items/metal_impact.ogg', 50, 1)
+			switch(pos_to)
```

Comment:
Why not just do
`var/old_pos = pos_from`
`pos_from = pos_to`
`pos_to = turn(old_pos, 180)`